### PR TITLE
Add deployment chapter

### PR DESCRIPTION
**Pull request by m1ghtym0**
I've moved the deployment into its own chapter and split it into logical parts.
This mostly based on the feedback of @dimakuv, see https://github.com/edgelesssys/marblerun/issues/147
Thanks again for going through this @dimakuv, great feedback for improving the documentation!:-)

### CODE REVIEW

**Comment by m1ghtym0** [comment]
@flxflx can we merge this?

File: content/docs/deployment/overview.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+To run Marblerun in the cloud, follow this [guide]({{< ref "docs/deployment/cloud.md" >}}). 
+
+To deploy Marblerun somewhere else, e.g., on-prem, use this [guide]({{< ref "docs/deployment/on-prem.md" >}}).
+
+## Kubernetes or standalone?
+
+Marblerun is designed to orchestrate confidential apps on Kubernetes. The simple deployment steps are described in this [guide]({{< ref "docs/deployment/kubernetes.md" >}}).
+
+Marblerun can also very well be used to manage confidential workloads indepently of Kubernetes, as is described in this [guide]({{< ref "docs/deployment/standalone.md" >}}).
```
Comment (katexochen) [inline content/docs/deployment/overview.md:21]: In my opinion, the "Overview" section does not provide many additional information compared to the headlines of the deployment guides. I think we can just give it a miss.

**Review by katexochen — COMMENTED** [review]
The sidebar is starting to get a bit messy and hard to navigate. We should improve it's logical structure and style, i.e. only expand subsections on hover/click. Maybe sth. for a new ticket, not this PR.

File: content/docs/deployment/overview.md (hunk 1 above)
Comment (m1ghtym0) [reply content/docs/deployment/overview.md:21]: I agree. However, I had the feeling without further explanation it is a bit unclear how those four "sections" of deployment fit together? Maybe it is and you're right we can dismiss it entirely.

**Pull request merged**